broker.go

```diff
@@ -156,6 +169,8 @@ func (b *Broker) Close() (err error) {
 	b.done = nil
 	b.responses = nil
 
+	atomic.StoreInt32(&b.opened, 0)
+
```

Comment:
Can we short-circuit `Close()` as well using this flag, instead of doing the `if b.conn == nil return ErrNotConnected` check (see above)? Not sure if this is useful.